src/models/user.ts

```diff
@@ -318,6 +320,38 @@ export class User extends Model {
     return new MSeriesFtpMeasurements(mSeriesFtpMeasurements, mSeriesFtpMeasurementsMeta, this.sessionHandler)
   }
 
+  async createMSeriesChallenge (params: {userLimit: number, name: string, challengeType: MSeriesChallengeType, focus: MSeriesChallengeFocus, startAt?: Date, endAt?: Date, goal?: number}) {
+    const { mSeriesChallenge } = await this.action('mSeriesChallenge:create', { ...params, isPublic: false }) as MSeriesChallengeResponse
+    return new PrivilegedMSeriesChallenge(mSeriesChallenge, this.sessionHandler)
+  }
+
+  async joinMSeriesChallenge (params: {joinCode: string}) {
+    const { mSeriesChallengeParticipant } = await this.action('mSeriesChallengeParticipant:create', { ...params, userId: this.id }) as MSeriesChallengeParticipantResponse
+    return new MSeriesChallengeParticipant(mSeriesChallengeParticipant, this.sessionHandler)
+  }
+
+  async getMSeriesChallenge (params: XOR<{id: number}, {joinCode: string}>) {
```

Comment:
Why have `getPrivilegedMSeriesChallenges` and `getJoinedMSeriesChallenges` as separate method calls when `getMSeriesChallenge` only has one call that returns deferent model types?

Reply:
I figured it was better practice and more clear to have two functions that each guarantee **one** returned type vs one function that has a union return type <PrivilegedMSeriesChallenge | JoinedMSeriesChallenge>

Subjective though I suppose, do you think it's better to just have one?

Reply:
I think it's better to be consistent. I would prefer the calls for `show` would adopt this same strategy (`getPrivilegedMSeriesChallenge`, `getJoinedMSeriesChallenge`, `getJoinableMSeriesChallenges`), but I'm assuming there are collisions that would prevent this from happening?

Reply:
Not exactly sure what you mean by collisions, but if you show a challenge using `id`, there is no way to know whether it's going to be a `JoinedMSeriesChallenge` or a `PrivilegedMSeriesChallenge` because it could potentially be either. So no matter what, `getMSeriesChallenge`  is going to have some kind of union return type.

Reply:
I would like to have `getPrivilegedMSeriesChallenge` be separate from `getMSeriesChallenge`.